pkg/ndctl/region.go

```diff
@@ -318,8 +318,9 @@ func (r *Region) namespaces(onlyActive bool) []*Namespace {
 	ndr := (*C.struct_ndctl_region)(r)
 
 	for ndns := C.ndctl_namespace_get_first(ndr); ndns != nil; ndns = C.ndctl_namespace_get_next(ndns) {
-		if !onlyActive || C.ndctl_namespace_is_active(ndns) == true {
-			namespaces = append(namespaces, (*Namespace)(ndns))
+		ns := (*Namespace)(ndns)
+		if (onlyActive && ns.Active()) || ns.Size() != 0 {
```

Comment:
Doesn't this break the semantic of `onlyActive`?

If it is true, then inactive namespaces are returned if their size is > 0, which is not what the caller asked for, is it?

Reply:
Yes, you are right, Fixed it now. Thanks.